sklearnex/tests/_utils.py

```diff
@@ -123,14 +123,24 @@ def gen_models_info(algorithms):
         else:
             raise KeyError(f"Unrecognized sklearnex estimator: {i}")
 
-        methods = set()
-        candidates = set(
-            [i for i in dir(est) if not i.startswith("_") and not i.endswith("_")]
-        )
-
-        for mixin, method, _ in mixin_map:
-            if issubclass(est, mixin):
-                methods |= candidates & set(method)
+        # remove BaseEstimator methods (get_params, set_params)
+        candidates = set(dir(est)) - set(dir(BaseEstimator))
+        # remove private methods
+        candidates = set([i for i in candidates if not i.startswith("_")])
+        # required to enable other methods
+        candidates = candidates - {"fit"}
```

Comment:
Why no fit?

Reply:
Good question, in most cases it is necessary to call fit in order to access further functionality, and it is assumed to have been used.